packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -542,197 +888,143 @@ const MessageButton: React.FC<{
-  showPostsYouveRead?: boolean;
-  showFrequentCommentedTopics?: boolean;
-};
 
-const UserTable: React.FC<UserTableProps> = ({
-  users,
-  classes,
-  gridClassName,
-  currentUser,
-  userDialogueChecks,
-  loadingNewDialogue,
-  createDialogue,
-  showBio,
-  showKarma,
-  showAgreement,
-  showPostsYouveRead,
-  showFrequentCommentedTopics,
-}) => {
+const DialogueUserRow = <V extends boolean>(props: DialogueUserRowProps<V> & { classes: ClassesType }): JSX.Element => {
+  const { targetUser, checkId, userIsChecked, userIsMatched, classes, currentUser, showKarma, showAgreement, showBio, showFrequentCommentedTopics, showPostsYouveRead } = props;
+  const { UsersName, DialogueCheckBox, MessageButton, MatchDialogueButton } = Components;
+
+  return <React.Fragment key={`${targetUser._id}_other`}>
+    <DialogueCheckBox
+      targetUserId={targetUser._id}
+      targetUserDisplayName={targetUser.displayName}
+      checkId={checkId}
+      isChecked={userIsChecked}
+      isMatched={userIsMatched}
+    />
+    <UsersName
+      className={classes.displayName}
+      documentId={targetUser._id}
+      simple={false} />
+    <MessageButton
+      targetUserId={targetUser._id}
+      currentUser={currentUser}
+    />
+    <MatchDialogueButton
+      isMatched={userIsMatched}
+      checkId={checkId}
+      targetUserId={targetUser._id}
+      targetUserDisplayName={targetUser.displayName}
+      currentUser={currentUser}
+    />
+    {showKarma && <div className={classes.centeredText}> {targetUser.total_power} </div>}
+    {showAgreement && <div className={classes.centeredText}> {targetUser.total_agreement} </div>}
+    {showBio && <UserBio
+      key={targetUser._id}
+      classes={classes}
+      userId={targetUser._id} />}
+    {showFrequentCommentedTopics && <UserTopTags
+      classes={classes}
+      targetUserId={targetUser._id} />}
+    {showPostsYouveRead && <UserPostsYouveRead
+      classes={classes}
+      targetUserId={targetUser._id}
+      limit={8} />}
+  </React.Fragment>;
+}
+
+const UserTable = <V extends boolean>(props: UserTableProps<V>) => {
+  const {
+    users,
+    classes,
+    gridClassName,
+    userDialogueChecks,
+    showHeaders,
+    ...rest
+  } = props;
 
-  const { UsersName } = Components;
+  const { DialogueUserRow } = Components;
 
   const headers = [
     "Dialogue",
     "Name",
     "Message",
     "Match",
-    ...(showKarma ? ["Karma"] : []),
-    ...(showAgreement ? ["Agreement"] : []),
-    ...(showBio ? ["Bio"] : []),
-    ...(showFrequentCommentedTopics ? ["Frequent commented topics"] : []),
-    ...(showPostsYouveRead ? ["Posts you've read"] : []),
+    ...(rest.showKarma ? ["Karma"] : []),
+    ...(rest.showAgreement ? ["Agreement"] : []),
+    ...(rest.showBio ? ["Bio"] : []),
+    ...(rest.showFrequentCommentedTopics ? ["Frequent commented topics"] : []),
+    ...(rest.showPostsYouveRead ? ["Posts you've read"] : []),
   ];
 
+  let rows;
+  if (props.isUpvotedUser) {
```

Comment:
needs to be completed for all cases and cleaned up

Reply:
I don't follow! And also @b0b3rt wrote this code

Reply:
Yes, I'll fix this bit

Reply:
@lagerros done